File: codingTest/programmers/python/가장 큰 정사각형 찾기.py
def solution(board):
    rowSize = len(board)
    colSize = len(board[0])

    if rowSize == 1 and colSize == 1:
        return board[0][0]
    else:
        maxValue = 0
        for i in range(1, rowSize):
            for j in range(1, colSize):
                if board[i][j] != 0:
                    board[i][j] = 1 + min(board[i-1][j], board[i-1][j-1], board[i][j-1])
                    maxValue = max(board[i][j], maxValue)
        return maxValue ** 2

File: codingTest/programmers/javascript/가장 큰 정사각형 찾기.js
function solution(board) {
  var rowSize = board.length;
  var colSize = board[0].length;

  if (rowSize == 1 && colSize == 1) return board[0][0];
  else {
    let max = 0;
    for (let i = 1; i < rowSize; i++) {
      for (let j = 1; j < colSize; j++) {
        // 상, 좌, 대각이 존재하는 0이 아닌 셀에서 최소값 구하기
        if (board[i][j]) {
          let min = Math.min(board[i - 1][j - 1], board[i - 1][j], board[i][j - 1]);
          // 최소값 + 1 를 현재 셀에 저장
          board[i][j] = min + 1;
          // 최대값 구하기
          max = Math.max(max, board[i][j]);
        }
      }
    }
    return max ** 2;
  }
}
